sklearn/preprocessing/discretization.py

```diff
@@ -52,6 +56,22 @@ class KBinsDiscretizer(BaseEstimator, TransformerMixin):
     dtype : number type, default=np.float
         Desired dtype of output.
 
+    strategy : {'uniform', 'quantile', 'kmeans'}, (default='uniform')
+        Strategy used to define the widths of the bins.
+
+        uniform
+            All bins in each feature have identical widths.
+        quantile
+            Widths are defined by a quantile transform, to have a uniform
+            number of samples in each bin.
+        kmeans
+            Widths are defined by a k-means on each features.
+
+    random_state : int, RandomState instance or None (default)
```

Comment:
Seeing as we're doing it in 1d, can't we just initialise kmeans at the midpoints of a uniform split, and make it deterministic?